github-actions/google-internal-tests/lib/main.ts

```diff
@@ -0,0 +1,79 @@
+import core from '@actions/core';
+import {context} from '@actions/github';
+import {Octokit} from '@octokit/rest';
+import minimatch from 'minimatch';
+
+async function main() {
+  if (context.repo.owner !== 'angular') {
+    core.info('Skipping Google Internal Tests action for non-Angular repos.');
+    return;
+  }
+  if (context.eventName !== 'pull_request_target' || context.payload.action !== 'synchronize') {
+    throw new Error(
+      'Expected Google Internal Tests action to be triggered for ' +
+        '`pull_request_target.synchronize` events.',
+    );
+  }
+
+  const githubToken = core.getInput('github-token', {required: true});
+  const rawPatterns = core.getInput('affected-file-patterns', {required: true});
+  const runTestGuideURL = core.getInput('run-tests-guide-url', {required: false});
+  const patterns = constructPatterns(rawPatterns);
+
+  const github = new Octokit({auth: githubToken});
+  const prNum = context.payload.pull_request!.number;
+  const prHeadSHA = context.payload.pull_request!.head!.sha;
+
+  const files = await github.paginate(github.pulls.listFiles, {
+    ...context.repo,
+    pull_number: prNum,
+  });
+
+  let affectsGoogle = false;
+  for (const f of files) {
+    if (patterns.some((p) => p.match(f.filename))) {
+      affectsGoogle = true;
+      break;
+    }
+  }
+
+  const status = affectsGoogle
+    ? ({
+        state: 'pending',
+        // The initial waiting status is expected to be overridden by the
+        // internal presubmit status service then.
+        description: `Waiting for Google Internal Tests. ${
+          runTestGuideURL ? `@Googlers: See Details for instructions -->` : ''
+        }`.trim(),
+        url: runTestGuideURL,
+      } as const)
+    : ({
+        state: 'success',
+        description: 'Does not affect Google.',
+      } as const);
+
+  await github.repos.createCommitStatus({
+    ...context.repo,
+    sha: prHeadSHA,
+    state: status.state,
```

Comment:
If we use `target_url` as the key for the status object, then you could just use `...status` here.

Reply:
good idea. done!